csrc/tensor_view.cpp

```diff
@@ -1476,6 +1476,21 @@ void TensorView::applyMmaSwizzle(MmaOptions options) {
   }
 }
 
+void TensorView::commitLeafToRFactor() {
+  TORCH_CHECK(
+      ir_utils::consumerTvsOf(this).empty(),
```

Comment:
Can you test if this error check actually works?

The use list of each tensor is not going to be updated unless one of its indirect consumers is registered as a fusion output. So, for example:

```
tv0 = makeSymbolicTensor;
tv1 = set(tv0);
tv2 = sum(tv1);
tv1->split(...);
tv1->commitLeafToRFactor();
```

I think this won't results in an error as none is registered as fusion outputs yet.

I think it's ok to start with limited error check as I'm still not clear how this interface is going to be used.

Reply:
Just checked. I think you are right. It won't raise an error unless you add output first...